service/controller/chart/v1/resource/release/desired.go

```diff
@@ -0,0 +1,135 @@
+package release
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/chart-operator/service/controller"
+	"github.com/giantswarm/chart-operator/service/controller/chart/v1/key"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	releaseName := key.ReleaseName(customObject)
+	tarballURL := key.TarballURL(customObject)
+
+	tarballPath, err := r.helmClient.PullChartTarball(ctx, tarballURL)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	defer func() {
+		err := r.fs.Remove(tarballPath)
+		if err != nil {
+			r.logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("deletion of %#q failed", tarballPath), "stack", fmt.Sprintf("%#v", err))
+		}
+	}()
+
+	chart, err := r.helmClient.LoadChart(ctx, tarballPath)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	configMapValues, err := r.getConfigMapValues(ctx, customObject)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	secretValues, err := r.getSecretValues(ctx, customObject)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
```

Comment:
The user values magic is also removed. This will move to app-operator. Here we just union the configmap and secret to pass a single set of values to Tiller.